<launch>
    <arg name="node_start_delay" default="60.0" /> 
    <arg name="nao_ip" default="$(optenv NAO_IP 10.0.1.230)" />
    <arg name="nao_port" default="$(optenv NAO_PORT 9559)" />

    <node pkg="fp_audio" type="audio_detected.py" name="audio_detected" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="fp_audio" type="speech2text.py" name="speech2text" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="fp_audio" type="manager.py" name="manager" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="fp_audio" type="video_detector.py" name="video_detector" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
     
    <node pkg="rasa_ros" name="rasa_action" type="rasa_action.sh" output="screen"/>
    <node pkg="rasa_ros" name="rasa_server" type="rasa_server.sh" output="screen"/> 
    <node pkg="rasa_ros" name="dialogue_server" type="dialogue_server.py" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>

    <node pkg="pepper_nodes" type="wakeup_node.py" name="wakeup_node" required="true" args="--ip=$(arg nao_ip) --port=$(arg nao_port)" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="pepper_nodes" type="text2speech_node.py" name="text2speech" required="true" args="--ip=$(arg nao_ip) --port=$(arg nao_port)" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="pepper_nodes" type="follow_node.py" name="follow_node" required="true" args="--ip=$(arg nao_ip) --port=$(arg nao_port)" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="pepper_nodes" type="image_input_node.py" name="image_input_node" required="true" args="--ip=$(arg nao_ip) --port=$(arg nao_port)" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>

</launch>
